docker-images/notebook/prepare.sh

```diff
@@ -4,6 +4,10 @@ set -x
 
 echo "Copy files from pre-load directory into home"
 cp --update -r -v /pre-home/. /home/jovyan
+rm -rf examples
```

Comment:
Is there some way to keep unrelated files in this directory around?  If a user copies an example I think it lands in the same directory.  Maybe we can only overwrite those files that are in the examples repository?

Reply:
You could conceivably clone to another directory and copy in (destructively) - but I think it's a good idea to have the `.git` be part of that, so that the history is there, and edits on the cluster could be committed back to the repo too.

Reply:
I agree that having `.git` be around would be helpful.  I suspect that there is probably a way to accomplish both tasks.

For example

```
mv examples tmp
git clone ...
cp --recursive --no-clobber tmp/* examples/ 
```

Reply:
Is there a downside that I'm not thinking of to use git for this:

```
git fetch origin
git reset --hard origin/master
git merge --strategy-option=theirs origin/master
```

This will update existing examples, and allow files not in origin/master to remain. 

It may not handle the case where the local HEAD is in front of origin/master.